compiler/rustc_interface/src/util.rs

```diff
@@ -441,7 +441,8 @@ pub fn get_codegen_sysroot(backend_name: &str) -> fn() -> Box<dyn CodegenBackend
 
     let mut file: Option<PathBuf> = None;
 
-    let expected_name = format!("rustc_codegen_{}", backend_name);
+    let expected_name =
+        format!("rustc_codegen_{}-{}", backend_name, release_str().expect("CFG_RELEASE"));
```

Comment:
I am a bit worried this is going to cause ICEs somewhere, but it shouldn't be in dist artifacts so I'm not *too* worried. We can deal with it if it comes up.